WalletWasabi.Fluent/Models/Wallets/WalletModel.cs

```diff
@@ -97,6 +97,8 @@ public WalletModel(Wallet wallet, IAmountProvider amountProvider)
 
 	public IAmountProvider AmountProvider { get; }
 
+	public TransactionFeeProvider TransactionFeeProvider => Wallet.TransactionFeeProvider;
```

Comment:
We shouldn't expose `TransactionFeeProvider` in the WalletModel.

The idea of having the models is to act as an encapsulation layer for all the APIs in the `WalletWasabi` project, via agnostic interfaces that operate with the UI object model.

This code:

```csharp
Fee = uiContext.AmountProvider.Create(transactionSummary.GetFee());
if (Fee is null || !Fee.HasBalance)
{
	Fee = uiContext.AmountProvider.Create(wallet.TransactionFeeProvider.GetFee(transactionSummary.GetHash()));
}
```

is business logic and depends on Client APIs. It should live in the Model rather than the ViewModel, to keep the ViewModels business logic free, and also to not leak implementation details in the Model interfaces

Reply:
Thank you, in 345a7b6 I got rid of business logic inside VMs.